huaweicloud/services/mrs/resource_huaweicloud_mapreduce_cluster.go

```diff
@@ -1007,13 +1006,62 @@ func updateMRSClusterNodes(ctx context.Context, d *schema.ResourceData, client *
 	return nil
 }
 
+func updateClusterName(d *schema.ResourceData, cfg *config.Config) error {
+	region := cfg.GetRegion(d)
+
+	var (
+		updateNameHttpUrl = "v2/{project_id}/clusters/{id}/cluster-name"
+		updateNameProduct = "mrs"
+	)
+	updateNameClient, err := cfg.NewServiceClient(updateNameProduct, region)
+	if err != nil {
+		return fmt.Errorf("error creating DWS Client: %s", err)
+	}
+
+	updateNamePath := updateNameClient.Endpoint + updateNameHttpUrl
+	updateNamePath = strings.ReplaceAll(updateNamePath, "{project_id}", updateNameClient.ProjectID)
+	updateNamePath = strings.ReplaceAll(updateNamePath, "{id}", d.Id())
+
+	updateNameOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+		MoreHeaders: map[string]string{
+			"Content-Type": "application/json;charset=UTF-8",
+		},
+	}
+
+	updateNameOpt.JSONBody = utils.RemoveNil(buildUpdateNameBodyParams(d))
+	_, err = updateNameClient.Request("PUT", updateNamePath, &updateNameOpt)
+	if err != nil {
+		return fmt.Errorf("error updating name of MRS cluster: %s", err)
+	}
+
+	return nil
+}
+
+func buildUpdateNameBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"cluster_name": utils.ValueIngoreEmpty(d.Get("name")),
```

Comment:
`name` is required, so I think IngoreEmpty is unnecessary.

Reply:
done